src/pycolmap/estimators/covariance.cc

```diff
@@ -57,14 +57,6 @@ void BindCovarianceEstimator(py::module& m) {
 
   using EstimatorBase = BundleAdjustmentCovarianceEstimatorBase;
   py::class_<EstimatorBase>(m, "BundleAdjustmentCovarianceEstimatorBase")
-      .def(
```

Comment:
I removed this custom method, because the same can be achieved by using one of the constructors.

Reply:
This feature was added in practical for the rig bundle adjuster. In such case one would like to initialize the estimator with a reconstruction and then set the pose blocks to different ones. Getting point blocks from reconstruction by iteration would be very slow in python unless we add a method in the Reconstruction class to support it.

Reply:
Why can you not use the constructor for this? I am not sure I see why this has to be done after creating the covariance class as opposed to during construction?

Reply:
Yes we can. Just that "Getting all point blocks from reconstruction by iteration would be very slow in python unless we add a method in the Reconstruction class to support it."

Reply:
OK, I see. But currently estimating covariance for points is anyway not supported, unless I misread the code?

Reply:
The point blocks is to set the points to be eliminated in Schur elimination, and it generally just solely comes from one reconstruction object.

Reply:
Sorry, I am probably not seeing the obvious. The SetPoseBlocks() only takes a vector of pose blocks but not point blocks. Why would one need to iterate over points of the reconstruction object in Python? I would expect that one would iterate over all images and extract the qvec/tvec arrays and pass them here. I would expect Python to be fast enough to iterate over a few hundreds or even thousands of images?

Reply:
Because if one uses the constructor, it will be Object(pose_blocks, point_blocks), where one needs to get the point_blocks. But with the SetPoseBlocks, one can do obj = Object(reconstruction); obj.set_pose_blocks(pose_blocks), which is fast in Python.

Reply:
> Sorry, I am probably not seeing the obvious. The SetPoseBlocks() only takes a vector of pose blocks but not point blocks. Why would one need to iterate over points of the reconstruction object in Python? I would expect that one would iterate over all images and extract the qvec/tvec arrays and pass them here. I would expect Python to be fast enough to iterate over a few hundreds or even thousands of images?

I m trying to say that SetPoseBlocks is useful here since from what I understand this PR attempted to remove this interface.

Reply:
Thanks for the patience. Got it finally :-)

How about the following: Only expose a single constructor and accepting an "image_ids" vector. If it is non-empty, it will only consider the specified images, otherwise it will extract all.

Reply:
For the rig case (especially without our new frame feature), generally we store the rig_from_world outside the reconstruction. And this was the reason why I developed this (pose_blocks, point_blocks) constructor and the ``SetPoseBlocks`` method, since there is no corresponding image ids for those data.

Reply:
In practical one can create a rig class and construct the bundle adjustment with pycolmap + pyceres, e.g. in lamar: https://github.com/microsoft/lamar-benchmark/blob/main/scantools/capture/session.py#L36. And this interface can perfectly support acquiring pose covariances in such case.